### PR TITLE
pmix: added check for pmix fence status

### LINKED ISSUE #5086: Check PMIx Fence return status in MPI_Init
One of the customers recently encountered the problem in the Slurm/PMIx environment.
For some unrelated reasons PMIx_Fence was cancelled by timeout by RM. However because OMPI doesn't checks the return status of the operation all processes proceeded further:
* modex: https://github.com/open-mpi/ompi/blob/v3.0.x/ompi/runtime/ompi_mpi_init.c#L663
* barrier: https://github.com/open-mpi/ompi/blob/v3.0.x/ompi/runtime/ompi_mpi_init.c#L840
I think this is not a correct behavior and processes should take an error code path in this case.

### PR DESCRIPTION
**Pull request by karasevb**
Closes #5086

### CODE REVIEW

**Comment by ibm-ompi** [comment]
The IBM CI (GNU Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/c7391e8f58032139aeb5325d6e23f29c

**Comment by ibm-ompi** [comment]
The IBM CI (XL Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/0fe463389af62da504c537cf69f68475

**Comment by ibm-ompi** [comment]
The IBM CI (PGI Compiler) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/1d1f14f27d9637dd6c832ac665a10a75

File: ompi/runtime/ompi_mpi_finalize.c [hunk 1]
```diff
@@ -257,7 +257,10 @@ int ompi_mpi_finalize(void)
              * communications/actions to complete.  See
              * https://github.com/open-mpi/ompi/issues/1576 for the
              * original bug report. */
-            opal_pmix.fence_nb(NULL, 0, fence_cbfunc, (void*)&active);
+            if (OMPI_SUCCESS != (ret = opal_pmix.fence_nb(NULL, 0, fence_cbfunc,
+                                                          (void*)&active))) {
+                OMPI_ERROR_LOG(ret);
+            }
```
Comment (rhc54) [inline ompi/runtime/ompi_mpi_finalize.c]: If the fence operation returns an error, then the ```wait_for_completion``` is never going to happen. You need to skip that step.

**Comment by hppritcha** [comment]
@karasevb we use the release stream tags on the underlying issue and for a PR into a non-master branch.

File: ompi/runtime/ompi_mpi_finalize.c (hunk 1 above)
Comment (karasevb) [reply ompi/runtime/ompi_mpi_finalize.c]: Thanks. Fixed.

**Pull request merged**